Error during bootloader compilation in Arch Linux

I am trying to compile the bootloader as I am packaging from a PKGBUILD, and I get this error, so I ran the commands manually to see if it was a fakeroot or environment error, and the problem persists. Previously, in 6.18, everything compiled without any problems, and I checked the changes in the code since that version, but I don't see anything that affects the bootloader. 

```
$ python -m build --wheel --no-isolation
No precompiled bootloader found or compile forced. Trying to compile the bootloader for you ...
Setting top to                           : /src/src/pyinstaller-6.19.0/bootloader 
Setting out to                           : /src/src/pyinstaller-6.19.0/bootloader/build 
Python Version                           : 3.14.2 (main, Jan  2 2026, 14:27:39) [GCC 15.2.1 20251112] 
MSVC target(s)                           : not found 
Checking for 'gcc' (C compiler)          : /usr/sbin/gcc 
Checking size of pointer                 : 8 
Platform                                 : Linux-64bit-intel detected based on compiler 
Checking for compiler flags -m64         : yes 
Checking for linker flags -m64           : yes 
Checking for compiler flags -Wno-error=unused-but-set-variable : yes 
Checking for header semaphore.h                                : yes 
Using semaphore API                                            : posix 
Checking for library dl                                        : yes 
Checking for library pthread                                   : yes 
Checking for library m                                         : yes 
Checking for library z                                         : yes 
Build tests                                                    : disabled 
Checking for header stdbool.h                                  : yes 
Checking for function unsetenv                                 : yes 
Checking for function mkdtemp                                  : yes 
Checking for function dirname                                  : yes 
Checking for function basename                                 : yes 
Checking for function wcsdup                                   : yes 
Checking for linker flags -Wl,--as-needed                      : yes 
Checking for program 'strip'                                   : /usr/sbin/strip 
'configure' finished successfully (1.265s)
'all' finished successfully (0.000s)
'distclean' finished successfully (0.001s)
Setting top to                           : /src/src/pyinstaller-6.19.0/bootloader 
Setting out to                           : /src/src/pyinstaller-6.19.0/bootloader/build 
Python Version                           : 3.14.2 (main, Jan  2 2026, 14:27:39) [GCC 15.2.1 20251112] 
MSVC target(s)                           : not found 
Checking for 'gcc' (C compiler)          : /usr/sbin/gcc 
Checking size of pointer                 : 8 
Platform                                 : Linux-64bit-intel detected based on compiler 
Checking for compiler flags -m64         : yes 
Checking for linker flags -m64           : yes 
Checking for compiler flags -Wno-error=unused-but-set-variable : yes 
Checking for header semaphore.h                                : yes 
Using semaphore API                                            : posix 
Checking for library dl                                        : yes 
Checking for library pthread                                   : yes 
Checking for library m                                         : yes 
Checking for library z                                         : yes 
Build tests                                                    : disabled 
Checking for header stdbool.h                                  : yes 
Checking for function unsetenv                                 : yes 
Checking for function mkdtemp                                  : yes 
Checking for function dirname                                  : yes 
Checking for function basename                                 : yes 
Checking for function wcsdup                                   : yes 
Checking for linker flags -Wl,--as-needed                      : yes 
Checking for program 'strip'                                   : /usr/sbin/strip 
'configure' finished successfully (1.184s)
'make_all' finished successfully (0.004s)
Waf: Entering directory `/src/src/pyinstaller-6.19.0/bootloader/build/debug'
[ 1/23] Compiling src/pyi_python.c
[ 2/23] Compiling src/pyi_global_win32.c
[ 3/23] Compiling src/pyi_dylib_python.c
[ 4/23] Compiling src/pyi_apple_events.c
[ 5/23] Compiling src/pyi_pyconfig_pep741.c
[ 6/23] Compiling src/pyi_multipkg.c
[ 7/23] Compiling src/pyi_dylib_tcltk.c
[ 8/23] Compiling src/pyi_launch.c
[ 9/23] Compiling src/pyi_pyconfig.c
[10/23] Compiling src/pyi_archive.c
[11/23] Compiling src/pyi_global_posix.c
[12/23] Compiling src/pyi_utils_win32.c
[13/23] Compiling src/pyi_exception_dialog.c
[14/23] Compiling src/pyi_path.c
[15/23] Compiling src/pyi_utils.c
[16/23] Compiling src/pyi_main.c
[17/23] Compiling src/pyi_utils_win32_low_level.c
[18/23] Compiling src/pyi_utils_posix.c
[19/23] Compiling src/pyi_splash.c
../../src/pyi_utils.c: In function ‘pyi_create_parent_directory_tree’:
../../src/pyi_utils.c:65:25: error: assignment discards ‘const’ qualifier from pointer target type [-Werror=discarded-qualifiers]
   65 |     for (subpath_cursor = strchr(filename, PYI_SEP); subpath_cursor != NULL; subpath_cursor = strchr(++subpath_cursor, PYI_SEP)) {
      |                         ^
cc1: all warnings being treated as errors

../../src/pyi_utils_posix.c: In function ‘_pyi_create_runtime_tmpdir’:
../../src/pyi_utils_posix.c:157:24: error: assignment discards ‘const’ qualifier from pointer target type [-Werror=discarded-qualifiers]
  157 |     for(subpath_cursor = strchr(runtime_tmpdir, '/'); subpath_cursor != NULL; subpath_cursor = strchr(++subpath_cursor, '/')) {
      |                        ^
cc1: all warnings being treated as errors

Waf: Leaving directory `/src/src/pyinstaller-6.19.0/bootloader/build/debug'
Build failed
 -> task in 'OBJECTS' failed with exit status 1 (run with -v to display more information)
 -> task in 'OBJECTS' failed with exit status 1 (run with -v to display more information)
ERROR: Failed compiling the bootloader. Please compile manually and rerun
```

--- a/bootloader/src/pyi_utils.c
+++ b/bootloader/src/pyi_utils.c
@@ -49,7 +49,7 @@ int
 pyi_create_parent_directory_tree(const struct PYI_CONTEXT *pyi_ctx, const char *prefix_path, const char *filename)
 {
     char path[PYI_PATH_MAX];
-    char *subpath_cursor;
+    const char *subpath_cursor;
     size_t path_length;
 
     /* Ensure that combined path length does not exceed max path. */

--- a/bootloader/src/pyi_utils_posix.c
+++ b/bootloader/src/pyi_utils_posix.c
@@ -134,7 +134,7 @@ static char *
 _pyi_create_runtime_tmpdir(const char *runtime_tmpdir)
 {
     char directory_tree_path[PYI_PATH_MAX];
-    char *subpath_cursor;
+    const char *subpath_cursor;
 
     /* Ensure runtime_tmpdir (and thus also its sub-path components)
      * do not exceed path limit. */

--- a/news/9371.bootloader.rst
+++ b/news/9371.bootloader.rst
@@ -1,0 +1,2 @@
+Fix errors when compiling with ``glibc`` 2.43 and compiler that defaults
+to using C23 standard.

